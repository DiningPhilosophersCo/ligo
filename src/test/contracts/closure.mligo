(* Test whether closures capture variables in CameLIGO *)

let test (k : int) : int =
  let j : int = k + 5 in
  let close : int -> int = fun (i : int) -> i + j in
  let j : int = 20 (* Shadow original variable *)
  in close 20
