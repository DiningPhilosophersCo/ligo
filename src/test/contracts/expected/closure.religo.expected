let test = (k: int): int => {
  let j: int = k + 5;
  let close: (int => int) = (i: int) => i + j;
  let j: int = 20;
  close(20)
};
